src/main/323C.c

```diff
@@ -44,15 +46,76 @@ void func_80012EB0(void)
 {
 }
 
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_80012EB8);
+void func_80012EB8(void)
+{
+    s32 var_a0;
+
+    if (D_80172203.unk0 == 0) { // g_GameVars.unk43
+        var_a0 = 0x4E;
+        if (D_801721F7 == 0) { // g_GameVars.unk37
+            var_a0 = 0x4B;
+        }
+    } else {
+        var_a0 = 0x4D;
+    }
+    func_80013AD8(var_a0, 0, 0);
+    func_80014A90(0, 0);
+    func_80013530();
+    D_80173C80 = D_8015D9C8;
+    D_80166BB4 = D_80142F70;
+    func_80015C10();
+}
 
 INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_80012F44);
 
 INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_80013014);
 
 INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_8001326C);
 
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_80013404);
+void func_80013404(u8 arg0)
+{
+    s32 temp_v1;
+    u8 var_s0;
+    s8* a0;
+    s8* var_v0;
+    struct EngineObj* ptr = &D_801721C0;
+
+    temp_v1 = *(s32*)0x1F800000;
+    *(void**)0x1F800100 = (temp_v1 * 0xA000 / 4) + &D_801499C8;
```

Comment:
I think it's more like this https://decomp.me/scratch/8Ki11
I suspect these are arrays of otags